core/src/main/java/org/elasticsearch/cluster/routing/allocation/RebalanceDecision.java

```diff
@@ -111,7 +123,7 @@ public String getExplanation() {
 
     @Override
     public void innerToXContent(XContentBuilder builder, Params params) throws IOException {
-        super.toXContent(builder, params);
+        builder.field("current_node_ranking", currentNodeRanking);
```

Comment:
we don't currently output the current node to which the shard is assigned.  would you like to add this?  this would only apply to RebalanceDecision and MoveDecision, and only the RebalanceDecision would have a "weight_ranking" on its current node.  This would be a more compelling reason to have each individual decision implement its own toXContent.